src/tools/atlasLocal/atlasLocalTool.ts

```diff
@@ -0,0 +1,155 @@
+import type { CallToolResult } from "@modelcontextprotocol/sdk/types.js";
+import type { TelemetryToolMetadata, ToolArgs, ToolCategory } from "../tool.js";
+import { ToolBase } from "../tool.js";
+import type { ToolCallback } from "@modelcontextprotocol/sdk/server/mcp.js";
+import type { Client } from "@mongodb-js/atlas-local";
+import { LogId } from "../../common/logger.js";
+import { z } from "zod";
+
+export const AtlasLocalToolMetadataDeploymentIdKey = "deploymentId";
+
+export abstract class AtlasLocalToolBase extends ToolBase {
+    public category: ToolCategory = "atlas-local";
+
+    protected verifyAllowed(): boolean {
+        return this.session.atlasLocalClient !== undefined && super.verifyAllowed();
+    }
+
+    protected async execute(...args: Parameters<ToolCallback<typeof this.argsShape>>): Promise<CallToolResult> {
+        const client = this.session.atlasLocalClient;
+
+        // If the client is not found, throw an error
+        // This should never happen:
+        // - atlas-local tools are only added after the client is set
+        //   this means that if we were unable to get the client, the tool will not be registered
+        // - in case the tool was registered by accident
+        //   verifyAllowed would still return false preventing the tool from being registered,
+        //   preventing the tool from being executed
+        if (!client) {
+            return {
+                content: [
+                    {
+                        type: "text",
+                        text: `Something went wrong on our end, this tool should have been disabled but it was not.
+please log a ticket here: https://github.com/mongodb-js/mongodb-mcp-server/issues/new?template=bug_report.yml`,
+                    },
+                ],
+                isError: true,
+            };
+        }
+
+        return this.executeWithAtlasLocalClient(client, ...args);
+    }
+
+    protected async lookupDeploymentId(client: Client, containerId: string): Promise<string | undefined> {
```

Comment:
```suggestion
    private async lookupDeploymentId(client: Client, containerId: string): Promise<string | undefined> {
```

This doesn't seem to be used outside of this file